src/config.cpp

```diff
@@ -716,46 +740,57 @@ CircuitConfig::CircuitConfig(const std::string& contents, const std::string& bas
     Parser parser(contents, basePath);
 
     _expandedJSON = parser.getExpandedJSON();
-
-    _components = parser.parseDefaultComponents();
+    _circuitType = parser.getCircuitType();
 
     _nodeSetsFile = parser.getNodeSetsPath();
 
     // Load node population overrides and check biophysical types
-    _nodePopulationProperties = parser.parseNodePopulations();
+    _nodePopulationProperties = parser.parseNodePopulations(_circuitType);
 
     // Load edge population overrides
-    _edgePopulationProperties = parser.parseEdgePopulations();
+    _edgePopulationProperties = parser.parseEdgePopulations(_circuitType);
+
+    Components components = parser.parseDefaultComponents();
 
     const auto updateDefaultProperties =
         [&](std::unordered_map<std::string, PopulationProperties>& map,
             const std::string& defaultType) {
             for (auto& entry : map) {
-                if (entry.second.type.empty()) {
-                    entry.second.type = defaultType;
+                auto& component = entry.second;
+                if (component.type.empty()) {
+                    component.type = defaultType;
                 }
-                if (entry.second.alternateMorphologyFormats.empty()) {
-                    entry.second.alternateMorphologyFormats = _components.alternateMorphologiesDir;
+
+                if (component.alternateMorphologyFormats.empty()) {
+                    component.alternateMorphologyFormats = components.alternateMorphologiesDir;
```

Comment:
I can't spot the difference between the `lhs` and `rhs`. Is it possible the `rhs` should be `_components.bio...`?